components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/dao/impl/IdentityUserMetadataMgtDAOImpl.java

```diff
@@ -0,0 +1,508 @@
+/*
+ * Copyright (c) 2023, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.recovery.dao.impl;
+
+import org.apache.commons.lang.StringUtils;
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.carbon.identity.core.util.IdentityDatabaseUtil;
+import org.wso2.carbon.identity.core.util.IdentityUtil;
+import org.wso2.carbon.identity.event.IdentityEventException;
+import org.wso2.carbon.identity.governance.internal.cache.IdentityDataStoreCache;
+import org.wso2.carbon.identity.governance.internal.cache.IdentityDataStoreCacheKey;
+import org.wso2.carbon.identity.governance.model.UserIdentityClaim;
+import org.wso2.carbon.identity.recovery.IdentityRecoveryConstants;
+import org.wso2.carbon.identity.recovery.dao.IdentityUserMetadataMgtDAO;
+import org.wso2.carbon.identity.recovery.internal.IdentityRecoveryServiceDataHolder;
+import org.wso2.carbon.user.api.RealmConfiguration;
+import org.wso2.carbon.user.core.UserCoreConstants;
+import org.wso2.carbon.user.core.UserStoreException;
+import org.wso2.carbon.user.core.UserStoreManager;
+import org.wso2.carbon.user.core.claim.ClaimManager;
+import org.wso2.carbon.user.core.service.RealmService;
+import org.wso2.carbon.user.core.util.DatabaseUtil;
+import org.wso2.carbon.user.core.util.UserCoreUtil;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.SQLException;
+import java.util.Locale;
+import java.util.Map;
+import javax.sql.DataSource;
+
+/**
+ * Implementation class for IdentityUserMetadataMgtDAO.
+ */
+public class IdentityUserMetadataMgtDAOImpl implements IdentityUserMetadataMgtDAO {
+
+    private static final Log log = LogFactory.getLog(IdentityUserMetadataMgtDAOImpl.class);
+
+    private final IdentityDataStoreCache identityDataStoreCache = IdentityDataStoreCache.getInstance();
+
+    @Override
+    public void updateUserMetadata(UserStoreManager userStoreManager, String username,
+                                   String claimURI, String value, String eventName) throws IdentityEventException {
+
+        if (username.contains(IdentityRecoveryConstants.TENANT_ASSOCIATION_MANAGER)) {
```

Comment:
Can the username be null here?

Reply:
Fixed in https://github.com/wso2-extensions/identity-governance/pull/697/commits/dd3419fd69cc40e847e40ed9dfb29c867a3f0018. A null check will be done before passing the username as a parameter for this method.